src/csvcubed/utils/sparql_handler/sparqlmanager.py

```diff
@@ -368,6 +374,33 @@ def select_codelist_cols_by_dataset_url(
     return map_codelist_cols_by_dataset_url_result(results)
 
 
+def select_codelist_primarykey_by_dataset_url(
+    rdf_graph: rdflib.ConjunctiveGraph, table_url: str
+) -> CodeListPrimaryKeyByDatasetUrlResult:
+    """
+    Queries the code list primary keyfor the given table url.
+
+    Member of :file:`./sparqlmanager.py`
+
+    :return: `CodeListPrimaryKeyByDatasetUrlResult`
+    """
+    results: List[ResultRow] = select(
+        _get_query_string_from_file(
+            SPARQLQueryName.SELECT_CODELIST_PRIMARY_KEY_BY_DATASET_URL
+        ),
+        rdf_graph,
+        init_bindings={"table_url": Literal(table_url)},
+    )
+
+    if len(results) != 1:
+        raise InvalidNumberOfRecordsException(
+            record_description=f"result for the {SPARQLQueryName.SELECT_CODELIST_PRIMARY_KEY_BY_DATASET_URL.value} sparql query",
+            excepted_num_of_records=1,
+            num_of_records=len(results),
+        )
+    return map_codelist_primary_key_by_dataset_url_result(results[0])
```

Comment:
This needs more thought about what happens when the primary key is a composite of multiple columns.